impl/client/src/main/javascript/web/dojo/pentaho/common/overrides/dijit/Dialog.js

```diff
@@ -0,0 +1,674 @@
+			// summary:
+			//		Call right before fade-in animation for new dialog.
+			//		Saves current focus, displays/adjusts underlay for new dialog,
+			//		and sets the z-index of the dialog itself.
+			//
+			//		New dialog will be displayed on top of all currently displayed dialogs.
+			//
+			//		Caller is responsible for setting focus in new dialog after the fade-in
+			//		animation completes.
+
+			// Save current focus
+			ds[ds.length - 1].focus = focus.curNode;
+
+			// Set z-index a bit above previous dialog
+			var zIndex = ds[ds.length - 1].dialog ? ds[ds.length - 1].zIndex + 2 : Dialog._DialogLevelManager._beginZIndex;
+			domStyle.set(dialog.domNode, 'zIndex', zIndex);
+
+			// Display the underlay, or if already displayed then adjust for this new dialog
+			DialogUnderlay.show(underlayAttrs, zIndex - 1);
+
+			ds.push({dialog: dialog, underlayAttrs: underlayAttrs, zIndex: zIndex});
+		},
+
+		hide: function(/*dijit/_WidgetBase*/ dialog){
+			// summary:
+			//		Called when the specified dialog is hidden/destroyed, after the fade-out
+			//		animation ends, in order to reset page focus, fix the underlay, etc.
+			//		If the specified dialog isn't open then does nothing.
+			//
+			//		Caller is responsible for either setting display:none on the dialog domNode,
+			//		or calling dijit/popup.hide(), or removing it from the page DOM.
+
+			if(ds[ds.length - 1].dialog == dialog){
+				// Removing the top (or only) dialog in the stack, return focus
+				// to previous dialog
+
+				ds.pop();
+
+				var pd = ds[ds.length - 1];	// the new active dialog (or the base page itself)
+
+				// Adjust underlay
+				if(ds.length == 1){
+					// Returning to original page.  Hide the underlay.
+					DialogUnderlay.hide();
+				}else{
+					// Popping back to previous dialog, adjust underlay.
+					DialogUnderlay.show(pd.underlayAttrs, pd.zIndex - 1);
+				}
+
+				// Adjust focus.
+				// TODO: regardless of setting of dialog.refocus, if the exeucte() method set focus somewhere,
+				// don't shift focus back to button.  Note that execute() runs at the start of the fade-out but
+				// this code runs later, at the end of the fade-out.  Menu has code like this.
+				if(dialog.refocus){
+					var focus;
+					if(dialog.dialogOpener){
+						focus = dialog.dialogOpener;
+						dialog.dialogOpener = null;
+					}else{
+            // If we are returning control to a previous dialog but for some reason
+            // that dialog didn't have a focused field, set focus to first focusable item.
+            // This situation could happen if two dialogs appeared at nearly the same time,
+            // since a dialog doesn't set it's focus until the fade-in is finished.
+            focus = pd.focus;
+            if(pd.dialog && (!focus || !dom.isDescendant(focus, pd.dialog.domNode))){
+              pd.dialog._getFocusItems(pd.dialog.domNode);
+              focus = pd.dialog._firstFocusItem;
```

Comment:
Where is the code where a menu item's parent would be followed until a stable anchor element to place focus on were found?

Reply:
Hi, Duarte. Turns out that code was not needed when setDialogOpener was introduced

Reply:
You can get away with dialog opener in every case, as long as you handle all instances explicitly — that was a given when we talked first.

I believe the goal would be to achieve a default behavior that would cover most cases, while having the dialog opener to cover the ones which the default behavior could not.

Is there any idea for how many cases are there that would benefit from this default behavior? Every dialog being opened from an Analyzer context menu?